src/core/HalfSwing.cpp

```diff
@@ -0,0 +1,222 @@
+/*
+ * HalfSwing.cpp - Swing algo that varies adjustments form 0-127
+ *              The algorythm is just the latter half of the HydrogenSwing groove..
+ *
+ * Copyright (c) 2004-2014 teknopaul <teknopaul/at/users.sourceforge.net>
+ *
+ * This file is part of LMMS - http://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+#include <QObject>
+#include <QDomElement>
+#include <QLabel>
+
+#include "Engine.h"
+#include "Groove.h"
+#include "HalfSwing.h"
+#include "Knob.h"
+#include "lmms_basics.h"
+#include "MidiTime.h"
+#include "Note.h"
+#include "Pattern.h"
+#include "Song.h"
+
+#include "stdio.h"
+
+
+HalfSwing::HalfSwing(QObject * _parent) :
+	QObject( _parent ),
+	Groove()
```

Comment:
If you aren't passing parameters into the `Groove` constructor, you can omit this `Groove()` bit - Groove will be default-initialized.